src/FirstSteps/RPG/Heroes/Hero.cs

```diff
@@ -37,81 +36,40 @@ public void DisplayStats()
             .AddItem("Agility", _agility, Color.Yellow)
             .AddItem("Health", _health, Color.Red)
             .AddItem("Damage", _damage, Color.Green));
-            Console.WriteLine($"Equipment:\ntotal price: {DisplayTotalPrice()}\ntotal items: {DisplayTotalItems()}\ntotal weight; {DisplayTotalWeight()} ");
+            Console.WriteLine($"Equipment:\ntotal price: {_equipment.DisplayTotalPrice()}\ntotal items: {_equipment.DisplayTotalItems()}\ntotal weight; {_equipment.DisplayTotalWeight()} ");
         }
+        protected abstract bool CanHandleSpecialItem(Item item);
 
         public bool AddItemToBackpack(Item item)
         {
             if (!CanHandleSpecialItem(item))
```

Comment:
You asked why items disappear from the inventory. Look at this:
![image](https://user-images.githubusercontent.com/10303508/222904605-6e45acfa-f213-4123-abbc-5bd272ee4aa6.png)
You always remove the item if the hero has enough coins. Maybe leave it like it is now, and I will create a new issue for this and you try to use `Exception` here.